interop/test_utils.go

```diff
@@ -684,98 +685,244 @@ func DoPickFirstUnary(ctx context.Context, tc testgrpc.TestServiceClient) {
-	start := time.Now()
-	client := tc
-	if resetChannel {
-		var conn *grpc.ClientConn
-		conn, err = grpc.Dial(serverAddr, dopts...)
+// SoakIterationResult represents the latency and status results of a single iteration in the soak test.
+type SoakIterationResult struct {
+	LatencyMs int64
+	Status    string // The status of the iteration
+}
+
+// ThreadResults stores the aggregated results for a specific thread during the soak test.
+type ThreadResults struct {
+	IterationsDone int
+	Failures       int
+	Latencies      *stats.Histogram
+}
+
+// ManagedChannel determines whether a new channel will be created or an existing one reused.
+type ManagedChannel func(*grpc.ClientConn) (*grpc.ClientConn, testgrpc.TestServiceClient)
+
+// createChannel initializes the shared channel (once for all threads).
+func createChannel(serverAddr string, dopts []grpc.DialOption) (*grpc.ClientConn, testgrpc.TestServiceClient) {
+	conn, err := grpc.NewClient(serverAddr, dopts...)
+	if err != nil {
+		log.Fatalf("Failed to create shared channel: %v", err)
+	}
+	client := testgrpc.NewTestServiceClient(conn)
+	return conn, client
+}
+
+func closeChannel(channel *grpc.ClientConn) {
+	if channel != nil {
+		err := channel.Close()
 		if err != nil {
-			return
+			log.Fatalf("Failed to close channel: %v", err)
 		}
-		defer conn.Close()
-		client = testgrpc.NewTestServiceClient(conn)
 	}
-	// per test spec, don't include channel shutdown in latency measurement
-	defer func() { latency = time.Since(start) }()
-	// do a large-unary RPC
+}
+
+// CreateNewChannel creates a new channel by shutting down the current one (for channel soak tests).
+func CreateNewChannel(currentChannel *grpc.ClientConn, serverAddr string, dopts []grpc.DialOption) (*grpc.ClientConn, testgrpc.TestServiceClient) {
+	closeChannel(currentChannel)
+	conn, client := createChannel(serverAddr, dopts)
+	return conn, client
+}
+
+// UseSharedChannel reuses the provided currentChannel.
+func UseSharedChannel(currentChannel *grpc.ClientConn) (*grpc.ClientConn, testgrpc.TestServiceClient) {
+	client := testgrpc.NewTestServiceClient(currentChannel)
+	return currentChannel, client
+}
+
+func doOneSoakIteration(
+	ctx context.Context,
+	client testgrpc.TestServiceClient,
+	soakRequestSize int,
+	soakResponseSize int,
+	copts []grpc.CallOption) (SoakIterationResult, error) {
+	start := time.Now()
+	var err error
+	// Do a large-unary RPC.
+	// Create the request payload.
 	pl := ClientNewPayload(testpb.PayloadType_COMPRESSABLE, soakRequestSize)
 	req := &testpb.SimpleRequest{
 		ResponseType: testpb.PayloadType_COMPRESSABLE,
 		ResponseSize: int32(soakResponseSize),
 		Payload:      pl,
 	}
+	// Perform the GRPC call.
 	var reply *testpb.SimpleResponse
 	reply, err = client.UnaryCall(ctx, req, copts...)
```

Comment:
Nit: use type inference instead:

```suggestion
	reply, err := client.UnaryCall(ctx, req, copts...)
```